python/setup.py

```diff
@@ -280,6 +280,7 @@ def get_packages(self):
         "aioredis < 2",
         "aiosignal",
         "frozenlist",
+        "requests",
```

Comment:
We are using requests + thread pool. Alternatively, we can use a library that supports async functionality (e.g., httpx). I prefer requests because it is really lightweight (100% written with built-in modules) and very stable (the project is basically "deprecated" and they will add no more code to this lib)

Reply:
Do we also need to change requirements/requirements.txt?

Also you can use pure urllib if you don't want extra dependency: https://stackoverflow.com/questions/25491541/python3-json-post-request-without-requests-library

Reply:
Thanks for the info! 

My impression is requests is pretty good. It looks like it has 0 external dependency. Maybe we can have more discussion about it when we make an official PR, and if it is not allowed, I can probably just write code myself (hopefully not)

Reply:
Also it looks like requests is already in `requirements.txt?`

Reply:
https://github.com/ray-project/ray/blob/826a8bb06cf918867392af7316805d7199735a38/python/requirements.txt#L27